src/App.tsx

```diff
@@ -26,14 +34,75 @@ const App: FunctionComponent<{}> = () => {
     setAllPokemon(parsedData);
   };
 
+  const toggleSearch = () => {
+    setSearch(!search);
+  };
+
+  const hideSearch = () => {
+    setFilter({ name: '', types: [] });
+    setSearch(false);
+  };
+
+  const renderSearch = () => {
+    if (search) {
+      return (
+        <SearchFilter
+          handleNameChange={nameFilter}
+          handleTypeChange={typeFilter}
+        />
+      );
+    } else {
+      return null;
+    }
+  };
+
+  const nameFilter = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setFilter({ ...filter, name: event.target.value.toLowerCase() });
+  };
+
+  const typeFilter = (selectedType: any) => {
+    let types = [];
+    if (selectedType) {
+      types = selectedType.map((type: any) => type.value);
+    }
+    setFilter({ ...filter, types: types });
+  };
+
+  const filterPokemon = () => {
+    let filteredPokemon = allPokemon;
+    if (filter) {
+      if (filter.name) {
+        filteredPokemon = filteredPokemon.filter(pokemon =>
+          pokemon.name.toLowerCase().includes(filter.name)
+        );
+      }
+
+      if (filter.types.length > 0) {
+        filteredPokemon = filteredPokemon.filter(pokemon => {
+          for (let i = 0; i < pokemon.types.length; i++) {
+            if (filter.types.includes(pokemon.types[i].type.name)) {
+              return true;
+            }
+          }
+          return false;
+        });
+      }
+    }
+    return filteredPokemon;
+  };
+
   return (
     <div className="App">
       <Router>
-        <Nav numberOfPokemon={allPokemon.length} />
+        <Nav numberOfPokemon={allPokemon.length} handleClick={toggleSearch} />
         {allPokemon.length > 0 ? (
           <Switch>
             <Route exact path="/pokemon">
-              <CardGrid allPokemon={allPokemon} />
+              {renderSearch()}
```

Comment:
_Slick_. So if `search` is falsey it just ignores the second argument of the Boolean operator. What always trips me out is that the whole thing evaluates to the truth_y_ value instead of `true`.